web_src/js/components/DiffCommitSelector.vue

```diff
@@ -0,0 +1,191 @@
+      if (this.lastReviewCommitSha) {
+        return this.commits.length - this.commits.findIndex((x) => x.id === this.lastReviewCommitSha) - 1;
+      }
+      return 0;
+    },
+    queryParams() {
+      return this.$el.parentNode.getAttribute('data-queryparams');
+    },
+    issueLink() {
+      return this.$el.parentNode.getAttribute('data-issuelink');
+    }
+  },
+  mounted() {
+    // Add a click listener to close this menu on click outside of this element
+    //  when the dropdown is currently visible opened
+    document.body.addEventListener('click', (event) => {
+      if (this.$el.contains(event.target)) return;
+      if (this.menuVisible) {
+        this.menuVisible = false;
+      }
+    });
+  },
+  methods: {
+    /** Opens our menu, loads commits before opening */
+    async toggleMenu() {
+      this.menuVisible = !this.menuVisible;
+      // load our commits when the menu is not yet visible (it'll be toggled after loading)
+      // and we got no commits
+      if (this.commits.length === 0 && this.menuVisible === true) {
+        this.isLoading = true;
+        await this.fetchCommits();
+        this.isLoading = false;
+      }
+    },
+    /** Load the commits to show in this dropdown */
+    async fetchCommits() {
+      const resp = await fetch(`${this.issueLink}/commits/list`);
+      const results = await resp.json();
+      this.commits.push(...results.commits);
+      this.commits.reverse();
+      this.lastReviewCommitSha = results.last_review_commit_sha !== '' ? results.last_review_commit_sha : null;
+      Object.assign(this.locale, results.locale);
+    },
+    showAllChanges() {
+      window.location = `${this.issueLink}/files${this.queryParams}`;
+    },
+    /** Called when user clicks on since last review */
+    changesSinceLastReviewClick() {
+      window.location = `${this.issueLink}/files/${this.lastReviewCommitSha}..${this.commits.at(-1).id}${this.queryParams}`;
+    },
+    /** Clicking on a single commit opens this specific commit */
+    commitClicked(commitId) {
+      window.location = `${this.issueLink}/commits/${commitId}${this.queryParams}`;
+    },
+    /**
+     * When a commit is clicked with shift this enables the range
+     * selection. Second click (with shift) defines the end of the
+     * range. This opens the diff of this range
+     * Exception: first commit is the first commit of this PR. Then
+     * the diff from beginning of PR up to the second clicked commit is
+     * opened
+     */
+    commitClickedShift(commit) {
+      this.hoverActivated = !this.hoverActivated;
+      commit.selected = true;
+      // Second click -> determine our range and open links accordingly
+      if (!this.hoverActivated) {
+        // find all selected commits and generate a link
+        if (this.commits[0].selected) {
+          // first commit is selected - generate a short url with only target sha
+          const lastCommit = this.commits.findLast((x) => x.selected);
```

Comment:
Should we perhaps add an extra check that redirects to `/files` when you select **all** commits?
While the current behavior is completely fine, I think it would be pretty user-friendly to redirect to the normal review page:
Evidently, the user wants to see every change and in other words should be able to approve it.

Reply:
nice idea - i'll add this check in the component.

Reply:
done.